composer/callbacks/checkpoint_saver.py

```diff
@@ -0,0 +1,439 @@
+        |                        | :func:`~composer.utils.dist.get_world_size`.          |
+        +------------------------+-------------------------------------------------------+
+        | ``{local_world_size}`` | The local world size, as returned by                  |
+        |                        | :func:`~composer.utils.dist.get_local_world_size`.    |
+        +------------------------+-------------------------------------------------------+
+        | ``{node_rank}``        | The node rank, as returned by                         |
+        |                        | :func:`~composer.utils.dist.getnode_rank`.            |
+        +------------------------+-------------------------------------------------------+
+        | ``{epoch}``            | The total epoch count, as returned by                 |
+        |                        | :meth:`~composer.core.time.Timer.epoch`.              |
+        +------------------------+-------------------------------------------------------+
+        | ``{batch}``            | The total batch count, as returned by                 |
+        |                        | :meth:`~composer.core.time.Timer.batch`.              |
+        +------------------------+-------------------------------------------------------+
+        | ``{batch_in_epoch}``   | The batch count in the current epoch, as returned by  |
+        |                        | :meth:`~composer.core.time.Timer.batch_in_epoch`.     |
+        +------------------------+-------------------------------------------------------+
+        | ``{sample}``           | The total sample count, as returned by                |
+        |                        | :meth:`~composer.core.time.Timer.sample`.             |
+        +------------------------+-------------------------------------------------------+
+        | ``{sample_in_epoch}``  | The sample count in the current epoch, as returned by |
+        |                        | :meth:`~composer.core.time.Timer.sample_in_epoch`.    |
+        +------------------------+-------------------------------------------------------+
+        | ``{token}``            | The total token count, as returned by                 |
+        |                        | :meth:`~composer.core.time.Timer.token`.              |
+        +------------------------+-------------------------------------------------------+
+        | ``{token_in_epoch}``   | The token count in the current epoch, as returned by  |
+        |                        | :meth:`~composer.core.time.Timer.token_in_epoch`.     |
+        +------------------------+-------------------------------------------------------+
+
+        The resulting checkpoint file name will be approximately
+        ``name_format_string.format(rank=..., local_rank=..., ...)``.
+
+        If :attr:`name_format_string` does not specify a file extension, ``.pt`` (if not using DeepSpeed)
+        or ``.tar`` (if using DeepSpeed) will be appended.
```

Comment:
I dont think we need to enforce an extension. We can default to ".pt" but users can use whichever extension they perfer. If they are using deepspeed, we just append, e.g. `checkpoint.pt.tar`